src/Database/Kioku/Internal/TrieIndex.hs

```diff
@@ -387,3 +434,19 @@ breakCommonPrefix b1 b2 =
      , BS.drop prefixLength b1
      , BS.drop prefixLength b2
      )
+
+breakCommonPrefixMultiKey :: MultiKey
+                          -> BS.ByteString
+                          -> (BS.ByteString, MultiKey, BS.ByteString)
+breakCommonPrefixMultiKey (MultiKey prefix kids) arc =
+  case breakCommonPrefix prefix arc of
+    (common, "", "") -> (common, MultiKey "" kids, "")
+    (common, remPrefix, "") -> (common, MultiKey remPrefix kids, "")
+    (common, "", remArc) ->
+      case dropWhile commonIsEmpty $ map (\k -> breakCommonPrefixMultiKey k remArc) kids of
```

Comment:
It's not entirely clear why `dropWhile` is sufficient here. Is it because we assume that at most one of the kids will have a non-empty common prefix with `remArc`?